linkerd/proxy/tap/src/grpc/server.rs

```diff
@@ -570,12 +572,15 @@ fn base_event<B, I: Inspect>(req: &http::Request<B>, inspect: &I) -> api::TapEve
             let mut m = api::tap_event::EndpointMeta::default();
             m.labels
                 .extend(labels.iter().map(|(k, v)| (k.clone(), v.clone())));
-            let tls = inspect.dst_tls(req);
-            let tls_status = TlsStatus::from(tls.as_ref());
-            m.labels.insert("tls".to_owned(), tls_status.to_string());
-            if let Conditional::Some(id) = tls {
-                m.labels
-                    .insert("server_id".to_owned(), id.as_ref().to_owned());
+            match inspect.dst_tls(req) {
+                Conditional::None(reason) => {
+                    m.labels.insert("tls".to_owned(), reason.to_string());
+                }
+                Conditional::Some(id) => {
+                    m.labels.insert("tls".to_owned(), "true".to_owned());
+                    m.labels
+                        .insert("server_id".to_owned(), id.as_ref().to_owned());
+                }
```

Comment:
Note: We've decoupled tap from metric labels (which is probably a good thing), so we replicate the prior tap-labeling behavior here.